cloud_Azure/terraform/module/examples/multiple_accounts_multiple_resource_group/profiles_populator.py

```diff
@@ -0,0 +1,428 @@
+        log.error("Failed to grant permissions to Service Principal")
+        return False
+    return True
+
+
+def consent_permissions(principal_id: str) -> bool:
+    COMMAND = f"ad app permission admin-consent --id {principal_id}"  # returns a dict
+    ATTEMPT_COUNT = 5
+    output_dict = az_cli(COMMAND, ATTEMPT_COUNT)
+    if not isinstance(output_dict, dict):
+        log.error("Failed to consent permissions to Service Principal")
+        return False
+    return True
+
+
+def save_profiles(file_path: str, profiles: List[AzureProfile]) -> bool:
+    config = configparser.ConfigParser()
+    for profile in profiles:
+        section = profile.name
+        config.add_section(section)
+        config[section]["subscription_id"] = profile.subscription_id
+        config[section]["tenant_id"] = profile.tenant_id
+        config[section]["principal_id"] = profile.principal_id
+        config[section]["principal_secret"] = profile.principal_secret
+        config[section]["location"] = profile.location
+        config[section]["resource_group_names"] = ",".join(profile.resource_group_names)
+        config[section]["storage_account_names"] = ""
+
+    try:
+        with open(file_path, "w", encoding="utf-8") as configfile:
+            config.write(configfile)
+        log.info("Saved profiles to '%s'", file_path)
+        return True
+    except OSError:
+        log.exception("Failed to save profiles to '%s'", file_path)
+        return False
+
+
+def profile_exists(profile_name: str, profiles: List[AzureProfile]) -> bool:
+    return find_profile(profile_name, profiles) is not None
+
+
+def find_profile(profile_name: str, profiles: List[AzureProfile]) -> Optional[int]:
+    for i, profile in enumerate(profiles):
+        if profile.name == profile_name:
+            return i
+    return None
+
+
+def find_secret(principal_id: str, profiles: List[AzureProfile]) -> Optional[str]:
+    for profile in profiles:
+        if profile.principal_id == principal_id and likely_valid_service_principal_secret(profile.principal_secret):
+            log.info("Secret for Service Principal ID '%s' found in profile '%s'", principal_id, profile.name)
+            return profile.principal_secret
+    log.info("Secret for Service Principal ID '%s' not found", principal_id)
+    return None
+
+
+def likely_valid_service_principal_secret(secret: str) -> bool:
+    SERVICE_PRINCIPAL_SECRET_LENGTH = 34
+    return len(secret) == SERVICE_PRINCIPAL_SECRET_LENGTH
+
+
+def insert_or_overwrite_profile(profiles: List[AzureProfile], profile: AzureProfile) -> None:
+    index = find_profile(profile.name, profiles)
+    if index is not None:
+        profiles[index] = profile
+    else:
+        profiles.append(profile)
+
+
+def backup_file(file_path: str) -> bool:
+    """Create file backup with time stamp suffix in the name"""
+
+    if not os.path.exists(file_path):
+        log.info("Source file '%s' doesn't exist. Skipping backup", file_path)
```

Comment:
Should be debug message.

Reply:
Logging levels to be clarified

Reply:
The tool should by default show to user only vital information, like:
- errors
- warnings
- progress indication for operations taking longer time
- names of resources created (especially if names are auto-generated), e.g. output files used
- any other information that the user may need to utilize the results of execution of the tool

All other messages should be logged at debug/verbose level.